waspc/cli/Command/Watch.hs

```diff
@@ -68,6 +58,50 @@ watch waspProjectDir outDir = FSN.withManager $ \mgr -> do
        in not (null filename)
             && not (take 2 filename == ".#") -- Ignore emacs lock files.
             && not (head filename == '#' && last filename == '#') -- Ignore emacs auto-save files.
-            && not (last filename == '~') -- Ignore emacs and vim backup files.
+            && last filename /= '~' -- Ignore emacs and vim backup files.
             && not (head filename == '.' && ".swp" `isSuffixOf` filename) -- Ignore vim swp files.
             && not (head filename == '.' && ".un~" `isSuffixOf` filename) -- Ignore vim undo files.
+
+recompile :: Path Abs (Dir Common.WaspProjectDir) -> Path Abs (Dir Lib.ProjectRootDir) -> IO ()
+recompile waspProjectDir outDir = do
+  waspSays "Recompiling on file change..."
+  compilationResult <- compileIO waspProjectDir outDir
+  case compilationResult of
+    Left err -> waspSays $ "Recompilation on file change failed: " ++ err
+    Right () -> waspSays "Recompilation on file change succeeded."
+  return ()
+
+listenForEvents :: Chan FSN.Event -> UTCTime -> IO () -> IO ()
+listenForEvents eventChan lastCompileTime recompilationFn = do
```

Comment:
I would rename `recompilationFn` to `doRecompile`, to reflect it is an IO action.